x-pack/plugins/lens/public/xy_visualization_plugin/xy_suggestions.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { partition } from 'lodash';
+import { Position } from '@elastic/charts';
+import { XYArgs } from './xy_expression';
+import { SuggestionRequest, VisualizationSuggestion, TableColumn } from '../types';
+
+/**
+ * Generate suggestions for the xy chart.
+ *
+ * @param opts
+ */
+export function getSuggestions(
+  opts: SuggestionRequest<XYArgs>
+): Array<VisualizationSuggestion<XYArgs>> {
+  return Object.entries(opts.tableColumns)
+    .filter(
+      ([_id, columns]) =>
+        // We require a column for x and one for y, and y needs numeric columns...
+        columns.length > 1 && columns.find(x => x.operation.dataType === 'number')
```

Comment:
The variable naming and comment here are confusing me. I think you are actually looking for something like "there is at least one numeric column", right?

```typescript
.filter(([_id, columns]) => (
  columns.length > 1 && columns.some(col => col.operation.dataType === 'number')
));
```

Reply:
I'm looking for there are at least 2 columns (so we can have an x and y) and at least one is numeric (for y). Definitely open to suggestions about how to make it clearer.